src/Parsers/TimeSpanParser.cs

```
@@ -0,0 +1,109 @@
+﻿using System.Globalization;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+using Remora.Commands.Parsers;
+
+namespace Octobot.Parsers;
+
+/// <summary>
+///     Parses <see cref="TimeSpan"/>s.
+/// </summary>
+[PublicAPI]
+public partial class TimeSpanParser : AbstractTypeParser<TimeSpan>
+{
+    private static readonly Regex Pattern = ParseRegex();
+
+    /// <summary>
+    ///     Parses <see cref="TimeSpan"/> from <see cref="string"/>.
+    /// </summary>
+    /// <returns>
+    ///     Parsed <see cref="TimeSpan"/>.
+    /// </returns>
+    /// <remarks>
+    ///     If parse wasn't successful, <see cref="TimeSpanParser"/> will return <see cref="TimeSpan.Zero"/>.
+    /// </remarks>
+    public static TimeSpan TryParse(string timeSpanString, CancellationToken ct = default)
+    {
+        if (timeSpanString.StartsWith('-'))
+        {
+            return TimeSpan.Zero;
```

This should return an error instead of Zero